app/models/plugins/thermometer.rb

```diff
@@ -28,4 +29,57 @@ def liquid_data
   def name
     self.class.name.demodulize
   end
+
+  def update_goal
+    self.goal = self.determine_next_goal
+    self.save
+  end
+
+  protected
+  def goal_should_update
+    current_total >= self.goal
+  end
+
+  def determine_next_goal
+    target_jump = self.determine_target_jump(self.current_total)
+
+
+    # This is slight overkill; it makes sure that when we increment the goal, it's targeting the closest jump value.
+    # Essentially, the goal here is to make sure that if we somehow don't update the goal right when it's on the next
+    # step (like 200 or 10000) we don't end up with an ugly goal like 301.
+    new_goal = current_total + target_jump
+    goal_target_difference = new_goal % target_jump
+    target_midpoint = target_jump * 0.5
+
+    if goal_target_difference == 0
+      new_goal
+    elsif goal_target_difference <= target_midpoint
+      new_goal - goal_target_difference
+    else
+      new_goal + target_jump - (goal_target_difference)
+    end
+  end
+
+  def determine_target_jump(count)
+    # We use a target jump here that's intended to bring us to round, pyschologically appealing
+    # numbers. People tend to react better seeing 25000 as a target than something like 22500, even
+    # if the latter is closer. So, this method defines a set of steps for the number to jump, based
+    # on the given value.
+    # The view rounds to the nearest thousand for the target, so don't have jumps of under 1000.
+    if count < 10000
```

Comment:
Yeah, I optimized this for clarity of logic. I could potentially change if there were strong feelings, but I'm down internet and on a brand new OSX (full wipe) install so it might take me a day or two before I could do that.